core/cmd/local_client.go

```diff
@@ -221,12 +223,12 @@ func (cli *Client) runNode(c *clipkg.Context) error {
 
 	lggr.Info("API exposed for user ", user.Email)
 
-	grp, grpCtx := errgroup.WithContext(rootCtx)
-
-	if err = app.Start(); err != nil {
+	if err = app.Start(rootCtx); err != nil {
 		return errors.Wrap(err, "error starting app")
```

Comment:
This context is cancelled in case of shutdown and this context is used only for the starting process. Once a service has started, it does not use this context anymore, therefore any goroutines spawned by the service are not controlled by this context.
However, a good point you made here - I think we need to trigger Stop() here, because we may have some services started and we need to stop them gracefully. I will revisit this logic today.

Reply:
Did you fix this @pinebit ?

Reply:
I revisited this, yes, and I think the main concern here was this: shall we try stopping subservices that have started already? I decided not to stop them and let the application just quit. See the comment above:
```
		// We do not try stopping any sub-services that might be started,
		// because the app will exit immediately upon return.
		// But LockedDB will be released by defer in above.
```
If you think we can do better here, please let me know.